### PR TITLE
proto+lnrpc: update gRPC-REST gateway to v1.8.6

### LINKED ISSUE #3168: Wallet recovery with passphrase is not working
initwallet REST API is working without passphrase but it fails with passphrase, with 'invalid passphrase' error message.

I opened first issue about this API #3128 (https://github.com/lightningnetwork/lnd/issues/3128) almost 10 days back also and i am testing these scenarios from almost last 15 days. I found that initwallet REST API is behaving strangly.

If I generate the genseed with passphrase through lncli and use this cipher_seed_mnemonic array to recover the wallet through initwallet REST API, then it works perfectly. But, if i will generate the genseed with passphrase through REST and use that cipher_seed_mnemonic array to recover with initwallet REST then i am getting 'invalid passphrase' error.

Your environment
version of lnd: 0.6.0-beta
which operating system: Windows
version bitcoind: v0.17.1

Steps to reproduce
1) Call GET /v1/genseed with aezeed_passphrase value.
2) Call POST /v1/initwallet with wallet_password. it will initialize the wallet successfully.
3) Delete the wallet manually. 
4) Call POST /v1/initwallet with wallet_password, aezeed_passphrase and cipher_seed_mnemonic array generated from step 1.

Initialize wallet with below details:
Method: POST,
url: https://192.168.1.8:8080/v1/initwallet,
payload: {
"wallet_password":"c3VoZWIxMjM=",
"cipher_seed_mnemonic":["absent","stuff","layer","holiday","slam","pelican","drift","heart","prepare","casual","treat","liar","small","venue","fragile","wise","frown","safe","brief","cherry","pigeon","pole","sound","rocket"],
"aezeed_passphrase":"dGhpc2lzbXlwYXNzcGhyYXNl"
}

Expected behaviour
It should successfully recover the wallet.

Actual behaviour
Failed response with { "error": "invalid passphrase", "code": 2 }

I have tested initwallet with multiple scenarios and all the scenarios are working with GRPC but not with REST.

### PR DESCRIPTION
**Pull request by guggero**
Fixes #3168

When using the REST gateway, fields that have the type `bytes` in `rpc.proto` cannot be used in query or path parameters (see #552 which lead to workarounds like #592).

The root cause of this is that the version of `grpc-gateway` we use doesn't support mapping to `[]byte` yet.
Trying to use `bytes` parameters results in `{"error":"unsupported field type uint8","code":3}`.
This has been fixed in https://github.com/grpc-ecosystem/grpc-gateway/pull/489.

Unfortunately, we can't use the newest version of `grpc-gateway` (which would be `1.11.3` at the moment) because that has a bug that messes up our `rpc.swagger.json` (https://github.com/grpc-ecosystem/grpc-gateway/issues/746).

After some trial and error, I found that version `1.8.6` is the newest version that works for our use case and produces the smallest diff in our generated code.

### CODE REVIEW

File: lnrpc/README.md [hunk 1]
```diff
@@ -170,11 +170,12 @@ $ cd $GOPATH/src/google.golang.org/genproto
...
 ```
 
-4. Install `grpc-ecosystem/grpc-gateway` at commit `f2862b476edcef83412c7af8687c9cd8e4097c0f`.
+4. Install `grpc-ecosystem/grpc-gateway` at version `v1.8.6`.
 ```bash
 $ git clone https://github.com/grpc-ecosystem/grpc-gateway $GOPATH/src/github.com/grpc-ecosystem/grpc-gateway
 $ cd $GOPATH/src/github.com/grpc-ecosystem/grpc-gateway
-$ git reset --hard f2862b476edcef83412c7af8687c9cd8e4097c0f
+$ git reset --hard v1.8.6
+$ go mod init
```
Comment (wpaulino) [inline lnrpc/README.md]: Why is this required? This version seems to support go modules.

Comment (guggero) [reply lnrpc/README.md]: It's not required. That was left over from a trial with an older version. Fixed.

**Comment by joostjager** [comment]
it would be nice if we can also mark all the string fields as deprecated now and remove them in 0.10

**Comment by wpaulino** [comment]
> it would be nice if we can also mark all the string fields as deprecated now and remove them in 0.10

@joostjager did you mean to leave this comment on https://github.com/lightningnetwork/lnd/pull/3647 instead?

**Comment by guggero** [comment]
@wpaulino no, he meant the fields introduced in #592 that worked around the bug that this PR fixes.

**Comment by halseth** [comment]
> > it would be nice if we can also mark all the string fields as deprecated now and remove them in 0.10
> 
> Do we want to do this? I think it would be nice to be able to remove the string fields at some point

Yep, can be done in separate PR though.

**Comment by joostjager** [comment]
I am getting a new error:

```
joost@joost-desktop $ make rpc
 Compiling protos.
cd ./lnrpc; ./gen_protos.sh
Generating root gRPC server protos
F1102 13:10:55.891383    8655 main.go:73] Cannot set flag paths=source_relative
--grpc-gateway_out: protoc-gen-grpc-gateway: Plugin failed with status code 255.
```

**Comment by joostjager** [comment]
Nvm, needed to update dep
